hbase-common/src/main/java/org/apache/hadoop/hbase/KeyValue.java

```diff
@@ -1389,29 +1396,25 @@ public byte getFamilyLength(int foffset) {
    */
   @Override
   public int getQualifierOffset() {
-    return getQualifierOffset(getFamilyOffset());
+    return getQualifierInternalPosition(getFamilyOffset(), getFamilyLength());
   }
 
-  /**
-   * @return Qualifier offset
-   */
-  private int getQualifierOffset(int foffset) {
-    return foffset + getFamilyLength(foffset);
+  @Override
+  public int getQualifierInternalPosition(int famOffset, byte famLength) {
+    return famOffset + famLength;
```

Comment:
Is this not current?
famOffset + famLength +  Bytes.SIZEOF_BYTE = qualifierInternalPosition
famOffset + famLength = quelifierLengthPosition